server.js

```diff
@@ -1,23 +1,63 @@
 const express = require("express");
 const cors = require("cors");
-
+const bodyParser = require("body-parser");
 const app = express();
 
+let counter = 0;
+
 app.use(cors());
+app.use(express.json());
+app.use(bodyParser.urlencoded({ extended: true }));
 
 const welcomeMessage = {
-  id: 0,
-  from: "Bart",
   text: "Welcome to CYF chat system!",
 };
 
 //This array is our "data store".
 //We will start with one message in the array.
 //Note: messages will be lost when Glitch restarts our server.
-const messages = [welcomeMessage];
+let messages = [welcomeMessage];
 
 app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
-app.listen(process.env.PORT);
+app.get("/messages", (req, res) => {
+  res.json(messages);
+});
+
+app.post("/messages", (req, res) => {
+  counter++;
+
+  let newMessage = req.body;
+
+  let creatMessage = {
+    id: counter,
+    from: newMessage.from,
+    message: newMessage.text,
+  };
+  messages.push(creatMessage);
+
+  res.json("Form submitted successfully!");
+  console.log(messages);
+});
+
+app.get("/messages/:id", (req, res) => {
+  console.log(req.params, "----> request.params");
+
+  const findItem = Number(req.params.id);
+
+  const selectedMessage = messages.filter((mess) => mess.id === findItem);
+
+  res.status(200).json(selectedMessage);
+});
+
+app.delete("message/:id", (req, res) => {
+  const findItem = Number(req.params.id);
+  console.log(findItem);
+  messages = messages.filter((mess) => mess.id !== findItem);
+});
```

Comment:
you don't send back the response, plz think about what will happen on the client side